Core/src/RequestWrapper.php

```diff
@@ -154,25 +173,79 @@ public function __construct(array $config = [])
      */
     public function send(RequestInterface $request, array $options = [])
     {
-        $restOptions = isset($options['restOptions']) ? $options['restOptions'] : $this->restOptions;
-        $timeout = isset($options['requestTimeout']) ? $options['requestTimeout'] : $this->requestTimeout;
-        $backoff = $this->configureBackoff($options);
-
-        if ($timeout && !array_key_exists('timeout', $restOptions)) {
-            $restOptions['timeout'] = $timeout;
-        }
+        $retryOptions = $this->getRetryOptions($options);
+        $backoff = $this->configureBackoff(
+            $retryOptions['retries'],
+            $retryOptions['retryFunction'],
+            $retryOptions['delayFunction']
+        );
 
         try {
-            return $backoff->execute($this->httpHandler, [$this->applyHeaders($request), $restOptions]);
+            return $backoff->execute($this->httpHandler, [
+                $this->applyHeaders($request),
+                $this->getRequestOptions($options)
+            ]);
         } catch (\Exception $ex) {
             throw $this->convertToGoogleException($ex);
         }
     }
 
+    /**
+     * Deliver the request asynchronously.
+     *
+     * @param RequestInterface $request A PSR-7 request.
+     * @param array $options [optional] {
+     *     Request options.
+     *
+     *     @type float $requestTimeout Seconds to wait before timing out the
+     *           request. **Defaults to** `0`.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type callable $restRetryFunction Sets the conditions for whether or
+     *           not a request should attempt to retry.
+     *     @type callable $restDelayFunction Sets the conditions for determining
+     *           how long to wait between attempts to retry.
+     *     @type array $restOptions HTTP client specific configuration options.
+     * }
+     * @return PromiseInterface<ResponseInterface>
+     * @experimental The experimental flag means that while we believe this method
+     *      or class is ready for use, it may change before release in backwards-
+     *      incompatible ways. Please use with caution, and test thoroughly when
+     *      upgrading.
+     */
+    public function sendAsync(RequestInterface $request, array $options = [])
+    {
+        // Unfortunately, the current ExponentialBackoff implementation doesn't
```

Comment:
This isn't ideal, but should be acceptable until we can start making breaking changes. If anyone has an idea of a better way to move forward I'm all ears! :)